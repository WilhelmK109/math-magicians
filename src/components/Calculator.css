.button-wrapper {
  height: 4em;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  background-color: #5a5a5a;
  color: black;
  flex: 1;
  cursor: pointer;
}
