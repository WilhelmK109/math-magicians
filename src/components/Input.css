.display-input {
  height: 70px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-end;
  padding: 16px;
  font-size: 28px;
  font-weight: 600;
  background-color: gray;
  color: white;
  width: 567px;
  border-top-right-radius: 16px;
  border-top-left-radius: 16px;
}
