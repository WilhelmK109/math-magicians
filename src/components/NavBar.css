.nav-bar {
  display: flex;
  justify-content: space-between;
  width: 80%;
  margin-top: 1%;
}

.nav-bar h1 {
  font-size: 2rem;
}

.nav-items {
  list-style: none;
  display: flex;
  align-items: center;
  gap: 10px;
  font-size: 1.5rem;
  align-self: flex-end;
  height: 100%;
  margin-top: 0%;
}

.nav-items > li:not(:last-child) {
  padding-right: 8px;
  border-right: 2px solid black;
}