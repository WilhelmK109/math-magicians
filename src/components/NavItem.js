import { NavLink } from 'react-router-dom';

export default function NavItem() {
  return (
    <ul className="nav-items">
      <li><NavLink to="/">Home</NavLink></li>
      <li><NavLink to="calculator">Calculator</NavLink></li>
      <li><NavLink to="quote">Quote</NavLink></li>
    </ul>
  );
}
