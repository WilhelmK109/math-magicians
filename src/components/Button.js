import PropTypes from 'prop-types';
import './Calculator.css';

function Button({ symbol, onClick }) {
  return (
    <div
      className="button-wrapper"
      role="button"
      tabIndex={0}
      onClick={onClick}
      onKeyDown={(e) => {
        if (e.key === 'Enter' || e.key === ' ') {
          onClick();
        }
      }}
    >
      {symbol}
    </div>
  );
}

Button.propTypes = {
  symbol: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
