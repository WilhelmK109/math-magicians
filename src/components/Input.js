import React from 'react';
import './Input.css';
import PropTypes from 'prop-types';

function Input({ result }) {
  return (
    <span className="display-input">{result}</span>
  );
}

Input.propTypes = {
  result: PropTypes.string.isRequired,
};

export default Input;
