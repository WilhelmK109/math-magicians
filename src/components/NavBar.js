import { Outlet } from 'react-router-dom';
import NavItem from './NavItem';
import './NavBar.css';

export default function NavBar() {
  return (
    <>
      <nav className="nav-bar">
        <h1>Math Magicians</h1>
        <NavItem />
      </nav>
      <Outlet />
    </>
  );
}
