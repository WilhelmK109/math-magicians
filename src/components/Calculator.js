import Button from './Button';

function Calculator() {
  return (
    <div className="App">
      <div className="calc-wrapper">
        <div className="calc-display">
          <Button symbol="0" />
        </div>
        <div className="row">
          <Button symbol="AC" />
          <Button symbol="+/-" />
          <Button symbol="%" />
          <Button symbol="รท" />
        </div>
        <div className="row">
          <Button symbol="7" />
          <Button symbol="8" />
          <Button symbol="9" />
          <Button symbol="*" />
        </div>
        <div className="row">
          <Button symbol="4" />
          <Button symbol="5" />
          <Button symbol="6" />
          <Button symbol="-" />
        </div>
        <div className="row">
          <Button symbol="1" />
          <Button symbol="2" />
          <Button symbol="3" />
          <Button symbol="+" />
        </div>
        <div className="row grid">
          <Button symbol="0" />
          <Button symbol="." />
          <Button symbol="=" />
        </div>
      </div>
    </div>
  );
}

export default Calculator;
