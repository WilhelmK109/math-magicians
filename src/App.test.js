import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Calculator from './components/calculator.js';
import Home from './components/Home.js';
import Quote from './components/Quote.js';
import NavBar from './components/NavBar.js';
import NavItem from './components/NavItem.js';
import Button from './components/Button.js';
import Input from './components/Input.js';

it('snapshot test for Calculator', () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot test for Home', () => {
  const component = renderer.create(<Home />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot test for Quote', () => {
  const component = renderer.create(<Quote />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot test for NavBar', () => {
  const component = renderer.create(
    <Router>
      <NavBar />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot test for NavItem', () => {
  const component = renderer.create(
    <Router>
      <NavItem />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot test for Button', () => {
  const component = renderer.create(
    <Button symbol="+" onClick={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot test for Input', () => {
  const component = renderer.create(
    <Input result="Test Result" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
