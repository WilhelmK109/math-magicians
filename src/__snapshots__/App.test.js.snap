// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Button 1`] = `
<div
  className="button-wrapper"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  +
</div>
`;

exports[`snapshot test for Calculator 1`] = `
<div
  className="App"
>
  <div
    className="calc-wrapper"
  >
    <div
      className="calc-display"
    >
      <span
        className="display-input"
      >
        0
      </span>
    </div>
    <div
      className="row"
    >
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        AC
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        +/-
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        %
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        รท
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        7
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        8
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        9
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        x
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        4
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        5
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        6
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        1
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        2
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        3
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        +
      </div>
    </div>
    <div
      className="row grid"
    >
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        0
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        .
      </div>
      <div
        className="button-wrapper"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot test for Home 1`] = `
<div
  className="home"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
     
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor expedita vitae officiis voluptatibus, voluptas libero doloribus incidunt numquam facere accusantium aliquid eligendi eum adipisci totam rerum, quis fugit quaerat vero enim exercitationem minus in dolores. Officia nobis doloribus voluptatem sit qui, recusandae iusto repellat molestias ullam corporis eos quod dignissimos. Dolor expedita vitae officiis voluptatibus, voluptas libero doloribus incidunt numquam facere accusantium aliquid eligendi eum adipisci totam rerum.
  </p>
  <p>
     
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor expedita vitae officiis voluptatibus, voluptas libero doloribus incidunt numquam facere accusantium aliquid eligendi eum adipisci totam rerum, quis fugit quaerat vero enim exercitationem minus in dolores. Officia nobis doloribus voluptatem sit qui, recusandae iusto repellat molestias ullam corporis eos quod dignissimos. Dolor expedita vitae officiis voluptatibus, voluptas libero doloribus incidunt numquam facere accusantium aliquid eligendi eum adipisci totam rerum.
  </p>
</div>
`;

exports[`snapshot test for Input 1`] = `
<span
  className="display-input"
>
  Test Result
</span>
`;

exports[`snapshot test for NavBar 1`] = `
<nav
  className="nav-bar"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="nav-items"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`snapshot test for NavItem 1`] = `
<ul
  className="nav-items"
>
  <li>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      className=""
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
  </li>
  <li>
    <a
      className=""
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </li>
</ul>
`;

exports[`snapshot test for Quote 1`] = `
<div
  className="display-quote"
>
  <div>
    Loading...
  </div>
</div>
`;
